src/resqml2_2/AbstractColorMap.h

```diff
@@ -0,0 +1,141 @@
+/*-----------------------------------------------------------------------
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"; you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-----------------------------------------------------------------------*/
+#pragma once
+
+#include "common/AbstractObject.h"
+
+namespace RESQML2_2_NS
+{
+	class AbstractColorMap : public COMMON_NS::AbstractObject
+	{
+	private:
+		/**
+		 * @param colorIndex	index of a color in the color map. It is cast to unsigned int in the case of a discrete color map 
+		 */
+		virtual gsoap_eml2_2::resqml2__HsvColor* getColor(double colorIndex) const = 0;
+
+	protected:
+		/**
+		 * Only to be used in partial transfer context
+		 */
+		AbstractColorMap(gsoap_resqml2_0_1::eml20__DataObjectReference* partialObject) : COMMON_NS::AbstractObject(partialObject) {}
```

Comment:
I think that a virtual destructor (empty would be good to have since this is a super class.

Reply:
Done